### ISSUE TITLE
[models] Mesh Bounding Box Does Not Take Model Transformation Into Account

### ISSUE THREAD

**Issue by dontpanic5** [raysan5/raylib#2956]
### Issue description

I'm making a game where my character flies around, so rotations and collision detection will play an important role. I've noticed that generated bounding boxes from models do not take a model's transformation into account.
This is noted in the source code at https://github.com/raysan5/raylib/blob/master/src/rmodels.c#L3286
I'm happy to look into making an improvement so that transformations are used, but wanted to make an issue first so I can make sure I'm approaching it the right way.

The comment says the minVertex and maxVertex should be transformed, but couldn't this create a bounding box that doesn't fully capture the area a bounding box would need to? Imagine if after a rotation is applied, a different vertex becomes what should be the minVertex. But, if I rotate all vertexes before calculating the minVertex and maxVertex that could be too expensive. So I'm curious which approach sounds better.

**Comment by JeffM2501** [comment]
Well a mesh itself doesn't have a transform, a model does. A mesh/model may be drawn at any location or in any orientation.

GetMeshBoundingBox's job is to just give you back the bbox of the mesh. only you know how you would use that mesh, so it's your job to transform that bbox into whatever space works for you. Some uses may want it local, some may want it in some other world space. It's not this function's job to handle that transformation, you have the math to do that yourself.

Basically it's not a world space function, it's just local to the mesh, the rest is up to you.  The function is working as designed.

**Comment by dontpanic5** [comment]
Since meshes don't have transformations, I can understand how GetMeshBoundingBox is reasonable in its current form. My concern is more with GetModelBoundingBox.
GetMeshBoundingBox is the basis for GetModelBoundingBox, and I believe that if a rotation is to be applied in the model's transformation the returned bbox could end up not useful even if transformed by the same transformation that is to be applied to the model. Not having the transformation applied would be unexpected since the model parameter has its transformation set when passed.
Consider a model in the shape of a cube which is aligned to the axes. And consider that it has a transformation which will rotate it 45 degrees about the z axis. The bounding box will be created based on the not transformed cube and the min and max will be the expected corners of the cube. But then if this bbox is rotated it actually then will represent a plane, as the min and max have been brought to the same x position. If the bbox is not rotated, it will not encompass the full space that the model occupies.

**Comment by JeffM2501** [comment]
Ahh, I'm sorry, I thought that was removed, my bad

**Comment by raysan5** [comment]
>  if a rotation is to be applied in the model's transformation the returned bbox could end up not useful

Yeah, that it. raylib `BoundingBox` and related functions are only intended for basic AABB, no rotations are considered. Considering rotations would require a redesign and increasing all this system complexity. I'm afraid this is out-of-scope for raylib at the moment.

For advance Model/Mesh collision detection I recommend using a physics library like Bullet or ODE.
